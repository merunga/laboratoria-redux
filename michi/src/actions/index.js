export const actionTypes = {
  SELECT: 'SELECT',
  RESET: 'RESET',
}

export const select = (x, y, turno) => ({
  type: actionTypes.SELECT,
  x,
  y,
  turno,
})

export const reset = () => ({
  type: actionTypes.RESET,
})
