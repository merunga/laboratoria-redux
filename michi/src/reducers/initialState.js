export default {
  michi: [
    [null, null, null],
    [null, null, null],
    [null, null, null],
  ],
  // historia: {
  //   comienza: 'X',
  //   jugadas: [],
  // },
}
