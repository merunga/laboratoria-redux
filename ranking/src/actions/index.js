export const actionTypes = {
  UP_VOTE: 'UP_VOTE',
  DOWN_VOTE: 'DOWN_VOTE',
  PREV: 'PREV',
  NEXT: 'NEXT',
}

export const upVote = id => ({
  type: actionTypes.UP_VOTE,
  id,
})

export const downVote = id => ({
  type: actionTypes.DOWN_VOTE,
  id,
})

export const prev = () => ({
  type: actionTypes.PREV,
})

export const next = () => ({
  type: actionTypes.NEXT,
})
