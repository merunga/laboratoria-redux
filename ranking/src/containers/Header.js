import { connect } from 'react-redux'
import Component from '../components/Header'

const Header = connect(
  state => ({
    seleccionado: state.canciones.find(c => c.id === state.seleccionado) || null,
  }),
  // mapDispatchToProps
)(Component)

export default Header
