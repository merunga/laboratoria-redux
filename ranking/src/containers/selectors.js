export const canciones = state => state.canciones.sort((c1, c2) => c1.votos <= c2.votos)

export const seleccionado = state => state.canciones.find(c => c.id === state.seleccionado) || null
