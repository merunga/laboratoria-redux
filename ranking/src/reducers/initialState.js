export default {
  canciones: [
    {
      id: 1, titulo: 'Psycho', autor: 'Post Malone', votos: 70,
    },
    {
      id: 2, titulo: 'Perfect', autor: 'Ed Sheeran', votos: 90,
    },
    {
      id: 3, titulo: 'God\'s Plan', autor: 'Drake', votos: 100,
    },
    {
      id: 4, titulo: 'Finesse', autor: 'Bruno Mars', votos: 80,
    },
    {
      id: 5, titulo: 'Meant To Be', autor: 'Bebe Rexha & Florida Georgia Line', votos: 60,
    },
  ],
  seleccionado: 1,
}
