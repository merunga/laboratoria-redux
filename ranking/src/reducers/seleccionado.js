import { actionTypes } from '../actions'

export default (state, action) => {
  switch (action.type) {
    case actionTypes.PREV:
    case actionTypes.NEXT: {
      const { canciones } = action
      const seleccionado = canciones.find(c => c.id === state) || null
      let selIdx = canciones.indexOf(seleccionado)
      switch (action.type) {
        case actionTypes.PREV:
          selIdx = selIdx > 0 ? selIdx - 1 : canciones.length - 1
          break
        case actionTypes.NEXT:
          selIdx = selIdx < canciones.length - 1 ? selIdx + 1 : 0
          break
        default:
          break
      }
      return canciones[selIdx].id
    }
    case actionTypes.SELECT:
      return action.id
    default:
      break
  }
  return state || null
}
