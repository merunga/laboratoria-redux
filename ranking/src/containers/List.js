import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import * as selectors from './selectors'
import Component from '../components/List'
import {
  upVote as onUpVote,
  downVote as onDownVote,
  select as onSelect,
} from '../actions'

const List = connect(
  state => ({
    canciones: selectors.canciones(state),
  }),
  dispatch => bindActionCreators({ onUpVote, onDownVote, onSelect }, dispatch),
)(Component)

export default List
