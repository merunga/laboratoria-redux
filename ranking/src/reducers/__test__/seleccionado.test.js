import initialState from '../initialState'
import seleccionado from '../seleccionado'
import * as actions from '../../actions'
import * as selectors from '../../containers/selectors'

const canciones = selectors.canciones(initialState)

describe('seleccionado', () => {
  it('prev', () => {
    expect(seleccionado(initialState.seleccionado, actions.prev(canciones))).toMatchSnapshot()
  })
  it('next', () => {
    expect(seleccionado(initialState.seleccionado, actions.next(canciones))).toMatchSnapshot()
  })
  it('select', () => {
    expect(seleccionado(initialState.seleccionado, actions.select(5, canciones))).toMatchSnapshot()
  })
})
