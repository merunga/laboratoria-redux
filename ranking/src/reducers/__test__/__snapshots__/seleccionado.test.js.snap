// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seleccionado next 1`] = `5`;

exports[`seleccionado prev 1`] = `4`;

exports[`seleccionado select 1`] = `5`;
